aadhar_demo_auth/data.py

```diff
@@ -0,0 +1,161 @@
+from lxml import etree
+import hashlib
+import base64
+from datetime import datetime
+from M2Crypto import Rand
+
+from .encryptor import DemoAuthEncryptor
+
+
+class DemoAuthData:
+    """
+    Prep all data nodes for the final XML request
+    """
+    def __init__(self, cfg, uid, demo_details, lang, tid=None):
+        """
+        cfg: Config object generated by DemoAuthConfig setup
+        uid: aadhar number
+        tid: terminal id of the machine that generated the beneficiary details pid block
+        """
+        self._cfg = cfg
+        self._x509_cert = cfg.common.uid_cert_path
+        self._tid = tid or cfg.common.tid
+        self._uid = uid
+        self._demo_details = demo_details
+        self._lang = lang
+
+        # even if we are not using the various options for authentication we still need to set
+        # "n" for those in the request because they are mandatory fields.
+        # We then update to "y" for the ones that we want to use.
+        self._uses = {
```

Comment:
i think we don't anyone to be messing up with this mapping and is just for internal use. Exposed through a method just for someone to see what the mapping is if needed or just for read access. Not sure if thats not the right "python" way to do this. Is there a better way?

Reply:
I see.  As a property, people won't be able to assign new values to `.uses`, but it's a dictionary, which are mutable, so they will be able to modify it.  That is
```python
demo_auth_data.uses = "foo"  # this fails
demo_auth_data.uses["foo"] = "bar"  # this modifies `_uses`
```
I can't think of any great ways to prevent you from modifying a dict attr, unfortunately.  There are some libraries which provide dictionary subclasses which can't be mutated, but that'd also mean you'd have to have all the information available before creating it, and I'm not sure that's a possibility here.

Reply:
Does this look like the right way to approach this here?
https://stackoverflow.com/a/38864561
[commit](https://github.com/dimagi/aadhaar-demo-auth/pull/2/commits/a2d66f738c8e35cd32b0d5b0b9991f012af07ad0)

Reply:
I see you shared that mutable objects can be edited.. made the change to revert to single "_"